src/harness/testcases/WINNF_FT_S_GRA_testcase.py

```diff
@@ -997,3 +999,217 @@ def test_WINNF_FT_S_GRA_16(self):
+    grant_1 = {
+        'cbsdId': 'REMOVE',
+        'operationParam': {
+            'maxEirp': 10,
+            'operationFrequencyRange': {
+                'lowFrequency': 3620000000,
+                'highFrequency': 3630000000
+            }
+        }
+    }
+    # LowFrequency > HighFrequency, will have device_c CBSD ID
+    grant_2 = {
+        'cbsdId': 2,
+        'operationParam': {
+            'maxEirp': 10,
+            'operationFrequencyRange': {
+                'lowFrequency': 3640000000,
+                'highFrequency': 3630000000
+            }
+        }
+    }
+    # Partially outside CBRS band, will have device_e CBSD ID
+    grant_3 = {
+        'cbsdId': 3,
+        'operationParam': {
+            'maxEirp': 10,
+            'operationFrequencyRange': {
+                'lowFrequency': 3450000000,
+                'highFrequency': 3600000000
+            }
+        }
+    }
+    # CBSD not part of claimed PPA, but in the PPA zone; requests PAL channel
+    # will have device_f CBSD ID
+    grant_4 = {
+        'cbsdId': 4,
+        'operationParam': {
+            'maxEirp': 10,
+            'operationFrequencyRange': {
+                'lowFrequency': 3550000000,
+                'highFrequency': 3560000000
+            }
+        }
+    }
+    # Create the actual config.
+    devices = [device_a, device_b, device_c, device_e, device_f]
+    grant_requests = [grant_0, grant_1, grant_2, grant_3, grant_4,]
+    # Devices inside and part of the claimed PPA.
+    # Contains the index of devices in the devices list.
+    # Eg: 0 means devices[0] which is device_a
+    ppa_cluster_list = [0]
+    # Devices in the zone, but not part of PPA.
+    # Contains the index of devices in the devices list.
+    # Eg: 4 means devices[4] which is device_f
+    devices_in_ppa_zone = [4]
+    pal_low_frequency = 3550000000
+    pal_high_frequency = 3560000000
+    pal_record = json.load(
+        open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+    ppa_record = json.load(
+        open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+    ppa_record, pal_record = makePpaAndPalRecordsConsistent(
+        ppa_record, [pal_record], pal_low_frequency, pal_high_frequency,
+        devices[ppa_cluster_list[0]]['userId'])
+    # Move devices in ppa_cluster_list and devices_in_ppa_zone
+    # into the PPA zone
+    for device_index in ppa_cluster_list:
+      devices[device_index]['installationParam']['latitude'], devices[
+          device_index]['installationParam'][
+              'longitude'] = getRandomLatLongInPolygon(ppa_record)
+    for device_index in devices_in_ppa_zone:
+      devices[device_index]['installationParam']['latitude'], devices[
+          device_index]['installationParam'][
+              'longitude'] = getRandomLatLongInPolygon(ppa_record)
+
+    config = {
+        'registrationRequest': devices,
+        'conditionalRegistrationData': conditionals,
+        'ppa': {
```

Comment:
For N2 >= 0 PPAs, should PPAs be an array?

Reply:
Fixed.